modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TaskServiceImpl.java

```diff
@@ -142,6 +142,18 @@
 
     @Override
     @Transactional
+    public Set<TaskError> setEnabledOrDisabled(final Task task) {
+        LOGGER.info("Saving task: {} with ID: {}", task.getName(), task.getId());
+        Set<String> fieldsToUpdate = new HashSet<>();
+        fieldsToUpdate.add("enabled");
+
+        tasksDataService.updateFromTransient(task, fieldsToUpdate);
+
+        return taskValidator.validate(tasksDataService.findById(task.getId()));
```

Comment:
we should drop validation, we are updating only one field and retrieving instance and validate it can take too much time